### PR TITLE
add DriveTrain Simulation

### PR DESCRIPTION
**Pull request by Weq511762338**

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -30,7 +40,24 @@ public RobotContainer() {
...
+    @Override
+    public double getAsDouble() {
+      return m_xbox.getLeftY();
+    }
+  }
 
+  private class Turn implements DoubleSupplier{
+    @Override
+    public double getAsDouble() {
+      return m_xbox.getLeftX();
```
Comment (prensing) [inline src/main/java/frc/robot/RobotContainer.java:53]: Please switch this to the right joystick per our spec.

File: src/main/java/frc/robot/subsystems/DriveTrain.java [hunk 2]
```diff
@@ -1,14 +1,25 @@
...
+import edu.wpi.first.wpilibj.RobotBase;
+import edu.wpi.first.wpilibj.RobotController;
 import edu.wpi.first.wpilibj.SPI.Port;
 import com.kauailabs.navx.frc.AHRS;
 import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
+import edu.wpi.first.wpilibj.simulation.DifferentialDrivetrainSim;
+import edu.wpi.first.wpilibj.simulation.EncoderSim;
+import edu.wpi.first.wpilibj.simulation.SimDeviceSim;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj.smartdashboard.FieldObject2d;
```
Comment (prensing) [inline src/main/java/frc/robot/subsystems/DriveTrain.java:22]: Some of these new imports are not used. Please remove them.

**Review by prensing — CHANGES_REQUESTED** [review]
Jack, please remove the simgui JSON files from the checkin. Those are dependent on the user and machine. You probably should put them in the .gitignore file.

**Comment by Weq511762338** [comment]
Can you explain how to put them to into the gitignore file? Those files always are generated when I tried to run the simulation

**Comment by prensing** [comment]
Edit the file ".gitignore" and just list them. It is a simple text file. Each line is a file name, or file name pattern, which will be ignored.

**Review by coachFitz0807 — APPROVED** [review]
Looks good as a start.

**Pull request merged**